Error when builing Docker image

**Describe the bug**

When I try to build a Docker image, I get error:

 ```
=> [build-env 6/7] RUN go mod download                                                                                                                                                              31.7s 
 => ERROR [build-env 7/7] RUN go build ./cmd/subfinder                                                                                                                                               10.4s 
------                                                                                                                                                                                                     
 > [build-env 7/7] RUN go build ./cmd/subfinder:                                                                                                                                                           
8.498 # github.com/projectdiscovery/fastdialer/fastdialer/metafiles                                                                                                                                        
8.498 /go/pkg/mod/github.com/projectdiscovery/fastdialer@v0.0.58/fastdialer/metafiles/shared.go:49:21: undefined: sync.OnceFunc                                                                            
8.498 /go/pkg/mod/github.com/projectdiscovery/fastdialer@v0.0.58/fastdialer/metafiles/shared.go:73:19: undefined: sync.OnceFunc
8.498 note: module requires Go 1.21
------
Dockerfile:8
--------------------
   6 |     WORKDIR /app/v2
   7 |     RUN go mod download
   8 | >>> RUN go build ./cmd/subfinder
   9 |     
  10 |     # Release
--------------------
ERROR: failed to solve: process "/bin/sh -c go build ./cmd/subfinder" did not complete successfully: exit code: 1

```
After changing Dockerfile as

```
$ git diff
diff --git a/Dockerfile b/Dockerfile
index 3d5f71f..99c6783 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -1,5 +1,5 @@
 # Build
-FROM golang:1.20.6-alpine AS build-env
+FROM golang:1.21-alpine AS build-env
```

The docker image compiles fine.

**Subfinder version**

```
$ docker run subfinder -version
[INF] Current Version: v2.6.5
[INF] Subfinder Config Directory: /root/.config/subfinder
```

**Complete command you used to reproduce this**
`docker build -t subfinder .`

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,5 +1,5 @@
 # Build
-FROM golang:1.20.6-alpine AS build-env
+FROM golang:1.21-alpine AS build-env
 RUN apk add build-base
 WORKDIR /app
 COPY . /app

--- a/README.md
+++ b/README.md
@@ -111,7 +111,7 @@ OPTIMIZATION:
 
 # Installation
 
-`subfinder` requires **go1.20** to install successfully. Run the following command to install the latest version:
+`subfinder` requires **go1.21** to install successfully. Run the following command to install the latest version:
 
 ```sh
 go install -v github.com/projectdiscovery/subfinder/v2/cmd/subfinder@latest

